tests/js/object-tests.js

```diff
@@ -515,5 +515,30 @@ module.exports = {
         TestCase.assertEqual(realm.objects('Date')[1].currentDate.getTime(), -10000);
         TestCase.assertEqual(realm.objects('Date')[2].currentDate.getTime(), 1000000000000);
         TestCase.assertEqual(realm.objects('Date')[3].currentDate.getTime(), -1000000000000);
+    },
+    
+    testInheritance: function() {
+        // the equivalent of `class InheritingObject extends Realm.Object`
+        function InheritingObject() {
```

Comment:
Personally I think we're fine with the "manually transpiled" code right now, but it is a good idea to add a native `class` declaration because it might have different semantics. I'll see how easy it would be to add an ES6-only test suite.